src/VestedRewardsDistributionJob.sol

```diff
@@ -0,0 +1,229 @@
+     */
+    function set(address dist, uint256 interval) external auth {
+        if (interval == 0) revert InvalidInterval();
+
+        if (!distributions.contains(dist)) distributions.add(dist);
+        intervals[dist] = interval;
+        emit Set(dist, interval);
+    }
+
+    /**
+     * @notice Removes `dist` from the job.
+     * @param dist The distribution contract.
+     */
+    function rem(address dist) external auth {
+        if (!distributions.remove(dist)) revert NotFound(dist);
+
+        delete intervals[dist];
+        emit Rem(dist);
+    }
+
+    /**
+     * @notice Checks if the job has the specified distribution contract.
+     * @param dist The distribution contract.
+     * @return Whether `dist` is set in the job or not.
+     */
+    function has(address dist) public view returns (bool) {
+        return distributions.contains(dist);
+    }
+
+    /**
+     * @notice Checks if the distribution is due for the specified contract.
+     * @param dist The distribution contract.
+     * @return Whether the distribution is due or not.
+     */
+    function due(address dist) public view returns (bool) {
+        // Gets the last time distribute() was called
+        uint256 last = VestedRewardsDistributionLike(dist).lastDistributedAt();
+        // If `last == 0` (no distribution so far), we allow it to be distributed immediately,
+        // otherwise, we can only distribute if enough time has elapsed since the last one.
+        if (last != 0 && block.timestamp < last + intervals[dist]) return false;
+
+        uint256 vestId = VestedRewardsDistributionLike(dist).vestId();
+        DssVestWithGemLike vest = VestedRewardsDistributionLike(dist).dssVest();
+        // Distribution is only due if there are unpaid tokens.
+        return vest.unpaid(vestId) > 0;
+    }
+
+    // --- Keeper Network Interface ---
+
+    /**
+     * @notice Executes the job though the keeper network.
+     * @param network The keeper identifier.
+     * @param args The arguments for execution.
+     */
+    function work(bytes32 network, bytes calldata args) external {
+        if (!sequencer.isMaster(network)) revert NotMaster(network);
+        if (args.length == 0) revert NoArgs();
+
+        (address dist) = abi.decode(args, (address));
+        // Prevents keeper from calling random contracts with a `distribute` method.
+        if (!has(dist)) revert NotFound(dist);
+        // Ensures that enough time has passed.
+        if (!due(dist)) revert NotDue(dist);
+
+        uint256 amount = VestedRewardsDistributionLike(dist).distribute();
+        emit Work(network, dist, amount);
+    }
+
+    /**
+     * @notice Checks if there is work to be done in the job.
+     * @param network The keeper identifier.
+     * @return ok Whether it should execute or not.
+     * @return args The args for execution.
+     */
+    function workable(bytes32 network) external override returns (bool ok, bytes memory args) {
```

Comment:
This works but it is too confusing. We keep both `workable()` and `work()` to not break the interface, but is seems to me that `work()` will rarely be used. Am i right?

Reply:
Both are used the same way as before:
- The keeper will call `workable` using something like [`ethers.js#staticCall`](https://docs.ethers.org/v6/api/contract/#BaseContractMethod-staticCall), which is pretty much like a transaction simulation, however the transaction is not actually submitted, so the state is not updated.
- If the `staticCall` returns `(true, <address>)`, then the keeper will call `work` with the proper arguments.

It's true that if the keeper `staticCall`s `workable()` and submit a transaction calling `workable()` again, it would have the same effect, however `workable()` is designed to look for work to be done and therefore is much more gas intensive.

Reply:
I see, so what you say is that there is no incentive to just  call `workable()` i.e. send an actual tx and waste gas...